pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -121,10 +115,16 @@ func (configgen *ConfigGeneratorImpl) BuildClusters(env *model.Environment, prox
 		if proxy.Type == model.Router && proxy.GetRouterMode() == model.SniDnatRouter {
 			outboundClusters = append(outboundClusters, configgen.buildOutboundSniDnatClusters(env, proxy, push)...)
 		}
+		// apply load balancer setting fot cluster endpoints
+		applyLocalityLBSetting(proxy.Locality, outboundClusters, env.Mesh.LocalityLbSetting)
 		outboundClusters = envoyfilter.ApplyClusterPatches(networking.EnvoyFilter_GATEWAY, proxy, push, outboundClusters)
 		clusters = outboundClusters
 	}
 
+	// Add a blackhole and passthrough cluster for catching traffic to unresolved routes
+	// DO NOT CALL PLUGINS for these two clusters.
+	clusters = append(clusters, buildBlackHoleCluster(env), buildDefaultPassthroughCluster(env, proxy))
```

Comment:
oh this needs to go to top as patch should be applicable to these as well..
the rule of the thumb is that nothing should be changed/added/removed after patch

Reply:
ok